proposals/0000-extract-runtime.md

```diff
@@ -0,0 +1,421 @@
+to create perfectly fit runtime variant they need for the application.**
+
+> __NOTE__: In this context we are talking about headless APIs, not related to UI.
+
+To do this, we should have an extensible layer, that can be built upon and configured, just like LEGO blocks. 
+
+For example, if the project is in need for the i18n API, and they are ready to accept the 
+drawbacks of increased app binary and increased app startup time, it should be easily doable, 
+without forking the React Native. The same can be said about other APIs, such as `crypto`, 
+where they might want to use our provided solution, not use it at all, or use in-house closed 
+source solution.
+
+The same can be said about changing the JS engine. The proposed layer would take care of allowing user
+to use another engine by using the already existing `jsi` interface. JS engine should be treated
+like a building block similar to others that can be changed easily by build time configuration.
+
+Right now, we have something similar already: the C++-only TurboModules, which are kind of tightly coupled 
+with the rest of the React Native and the distinction can be a bit confusing for newcomers.
+
+Such blocks would have access to Event Loop and JS engine, so new globals can be added in as-needed
+basis.
+
+Taking WebAPIs as an example, having their implementations split across multiple different TurboModules,
+we can release and version them independently further simplifying update process.
+In case the `react-natime-runtime` changes in a breaking manner, we can tightly couple the versions
+of such TurboModules or make them part of `react-native-runtime` that are extracted later, when
+the breaking changes become less frequent.
+
+__Secondary goal is to be able to develop and test those headless modules easily, on the CI pipeline without visual environment__.
+
+By having the possibility to run the "blocks" (C++ only turbo modules) in a headless JS environment we
+are able to test the implementations easily on many CI platforms and on different JS engines (ideally).
+
+A side effect of this is we can perhaps achieve faster build times both on CI and locally, resulting in lower cost of such pipelines.
+
+__Tertiary goal is to bring OOT (out of tree) platforms closer to first-class citizens__
+
+If a Turbo Module does not need access to UI in any way to provide its features, having it work with `react-native-runtime`
+rather than `react-native` would make it easier to maintain for currently OOT platforms (such as rn-windows and rn-mac) where
+the current `react-native` code cannot be re-used in 100% (needs further clarification). 
+Right now we are biased towards mobile platforms only.
+
+### Non-goals
+
+This RFC does **not** aim to:
+
+ - Replace or deprecate proposal [Move iOS and Android Specific Code to Their Own Packages](https://github.com/react-native-community/discussions-and-proposals/pull/49)
+
```

Comment:
I think we don't need to state that

Reply:
I wanted to highlight that this does not collide with mentioned RFC